### PR TITLE
Adding application methods

### PR DESCRIPTION
**Pull request by ahnaf-zamil**
Added application methods.

### Checklist

- [x] I have run `nox` and all the pipelines have passed. (MyPy was failing because I didn't have C++ build tools, but except that all other pipelines passed)
- [x] I have made unittests according to the code I have added/modified/deleted.

### Related issues
#33

### CODE REVIEW

File: hikari/applications.py [hunk 1]
```diff
@@ -613,6 +613,88 @@ def make_cover_image_url(self, *, ext: str = "png", size: int = 4096) -> typing.
...
         )
 
+    async def fetch_guild(self) -> typing.Optional[guilds.RESTGuild]:
+        """Fetch the guild this application is linked to if sold on Discord.
+
+        Parameters
+        ----------
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            The guild to fetch. This can be the object
+            or the ID of an existing guild.
```
Comment (ashwinvin) [inline hikari/applications.py:623]: This is not needed.

File: hikari/applications.py [hunk 2]
```diff
@@ -613,6 +613,88 @@ def make_cover_image_url(self, *, ext: str = "png", size: int = 4096) -> typing.
...
+        return None
+
+    async def fetch_guild_preview(self) -> typing.Optional[guilds.GuildPreview]:
+        """Fetch the preview of the guild this application is linked to if sold on Discord.
+
+        Parameters
+        ----------
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            The guild to fetch the preview of. This can be a
+            guild object or the ID of an existing guild.
```
Comment (ashwinvin) [inline hikari/applications.py:663]: This too is not needed

**Review by ashwinvin — CHANGES_REQUESTED** [review]

File: hikari/applications.py (hunk 1 above)
Comment (ahnaf-zamil) [reply hikari/applications.py:623]: Oops yeah my bad. Basically copied the docstring from rest.fetch_guild

**Review by ashwinvin — DISMISSED** [review]

**Comment by ahnaf-zamil** [comment]
All 3.10 pipelines are failing because the typing extensions are messed up, won't be a problem though. Will it?

**Review by FasterSpeeding — DISMISSED** [review]

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**